pandas/formats/style.py

```diff
@@ -987,6 +989,15 @@ def bar(self, subset=None, align='left', axis=0,
 
         if not(is_list_like(color)):
             color = [color, color]
+        elif len(color) == 1:
+            color = [color[0], color[0]]
+        elif len(color) > 2:
+            msg = ("Must pass `color` as string or a list-like"
+                   " of length 2 [ `color_negative`, `color_positive`] "
+                   "(eg: color=['#d65f5f', '#5fba7d'])\n"
+                   "Only the first two list elements will be used here.")
+            warnings.warn(msg, UserWarning)
```

Comment:
Should I do `raise TypeError(msg)` instead of issuing a warning?

(removing the line saying that only two elements will be used of course...)